Answer the channel before saying things (SayNumber, SayDigits,
SayCharacters, SayPhonetic).

--- a/main/pbx.c
+++ b/main/pbx.c
@@ -6060,6 +6060,9 @@ static int pbx_builtin_saynumber(struct 
 			return -1;
 		}
 	}
+	if (chan->_state != AST_STATE_UP) {
+	    ast_answer(chan);
+	}
 	return ast_say_number(chan, atoi(tmp), "", chan->language, options);
 }
 
@@ -6067,8 +6070,12 @@ static int pbx_builtin_saydigits(struct 
 {
 	int res = 0;
 
-	if (data)
+	if (data) {
+		if (chan->_state != AST_STATE_UP) {
+		    ast_answer(chan);
+		}
 		res = ast_say_digit_str(chan, data, "", chan->language);
+	}
 	return res;
 }
 
@@ -6076,8 +6083,12 @@ static int pbx_builtin_saycharacters(str
 {
 	int res = 0;
 
-	if (data)
+	if (data) {
+		if (chan->_state != AST_STATE_UP) {
+		    ast_answer(chan);
+		}
 		res = ast_say_character_str(chan, data, "", chan->language);
+	}
 	return res;
 }
 
@@ -6085,8 +6096,12 @@ static int pbx_builtin_sayphonetic(struc
 {
 	int res = 0;
 
-	if (data)
+	if (data) {
+		if (chan->_state != AST_STATE_UP) {
+		    ast_answer(chan);
+		}
 		res = ast_say_phonetic_str(chan, data, "", chan->language);
+	}
 	return res;
 }
 
